classes/item.rb

```diff
@@ -0,0 +1,51 @@
+# 01 Create Item class in a separate .rb file.
+require 'date'
+
+class Item
+  def initialize(date_published, id)
+    @date_published = Date.parse(date_published)
+    @archived = false
+    @id = id
+  end
+
+  # 02
+  # Add all methods visible in the diagram.
+  def add_author(author)
+    @author = author
+    author.add_item(self)
+  end
+
+  def add_source(source)
+    @source = source
+    source.add_item(self)
+  end
+
+  def add_label(label)
+    @label = label
+    label.add_item(self)
+  end
+
+  def add_genre(genre)
+    @genre = genre
+    genre.add_item(self)
+  end
+
+  # 03 Implement methods:
+  # can_be_archived?() in the Item class
+  # should return true if published_date is older than 10 years
+  # otherwise, it should return false
+  def can_be_archived?
+    true if (Date.today.to_time.year - @date_published.to_time.year) > 10
+  end
```

Comment:
- [ ] Please know that this method should return false when the condition isn't met, and easy way to do this on ruby is this:

```ruby
    def can_be_archived?
      today = Date.today
      10_years_ago = Date.new(today.year - 10, today.month, today.day)

      @published_date < 10_years_ago
    end
```